Sources/Promises/Promise.swift

```diff
@@ -40,14 +40,14 @@ public final class Promise<Value> {
   public convenience init(_ work: @autoclosure () throws -> Value) {
     do {
       let resolution = try work()
-      switch resolution {
-      case let error as NSError:
-        self.init(error)
-      case let objCPromise as ObjCPromise<AnyObject>:
-        self.init(objCPromise)
-      default:
-        self.init(ObjCPromise<AnyObject>.__resolved(with: Promise<Value>.asAnyObject(resolution)))
-      }
+      if type(of: resolution) is NSError.Type {
+          let error = resolution as! NSError
+          self.init(error)
+      } else if type(of: resolution) is ObjCPromise<AnyObject>.Type {
```

Comment:
Was this change necessary? If not prefer reverting to minimize the scope of the change.

Reply:
This can be undone you are right

Reply:
Update: I was wrong, this cannot be undone and it must stay

Reply:
To clarify, would this work:
```
      } else if let objCPromise = resolution as? ObjCPromise<AnyObject> {
        self.init(objCPromise)
```

This seems to be equivalent to the old code... the new code seems like it could be subtly different (I don't know enough about Swift to be sure)... so reducing the amount of change is preferred.